scattering/scattering.py

```diff
@@ -133,14 +133,14 @@ def structure_factor(trj, pair=None, Q_range=(0.5, 50), n_points=1000, framewise
                         N_i = len(top.select('element {}'.format(e1)))
                         N_j = len(top.select('element {}'.format(e2)))
 
-                        N_i_1 = len(top_1.select('element {}'.format(e1)))
-                        N_j_2 = len(top_2.select('element {}'.format(e2)))
+                        N_i_1 = n_element_1
+                        N_j_2 = n_element_2
 
                         scale = N_i_1 *  N_j_2 / (N_i * N_j)
                         Number_scale['{0}{1}'.format(e1, e2)] = scale
 
-                    integral = simps(r ** 2 * (g_r - 1) * np.sin(q * r) / (q * r), r)
-                    pre_factor = x_a * x_b * f_a * f_b * 4 * np.pi * rho * scale
+                    integral = simps(r ** 2 * (g_r - g_r[-1]) * np.sin(q * r) / (q * r), r) * scale
```

Comment:
I think you can keep this as `g_r - 1`.  I know in the paper the equation is `g_r - g_r(r -> infinity)` but `g_r(r->infinity)` should equal 1.  My worry is that if the `Q_range` is set to be bigger than the simulation box that this will lead to some weird results.

Reply:
Thanks, let me change it.